src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -613,26 +629,49 @@ private void processLogsAndUpload() throws InterruptedException {
                             if (componentCurrentProcessingLogFile.containsKey(componentName)) {
                                 CurrentProcessingFileInformation processingFileInformation =
                                         componentCurrentProcessingLogFile.get(componentName);
-                                if (processingFileInformation.fileName.equals(file.getAbsolutePath())
-                                        && processingFileInformation.lastModifiedTime == file.lastModified()) {
+                                //if (processingFileInformation.fileName.equals(file.getAbsolutePath())
+                                if (processingFileInformation.fileHash.equals(fileHash)
+                                        && processingFileInformation.lastModifiedTime <= file.lastModified()) {
                                     startPosition = processingFileInformation.startPosition;
                                 }
                             }
-                            LogFileInformation logFileInformation = LogFileInformation.builder()
-                                            .logFile(file)
-                                            .startPosition(startPosition)
-                                            .fileHash(fileHash)
-                                            .build();
+
+                            // If the file is not completed read or the file is recorded as
+                            // update the global map of hash and name pair
+                            fileHashNamePairs.putIfAbsent(fileHash, file.getAbsolutePath());
+                            if (!fileHashNamePairs.get(fileHash).equals(file.getAbsolutePath())) {
+                                fileHashNamePairs.replace(fileHash, file.getAbsolutePath());
+                            }
+
+                            // if the file is uploaded and it is not active file, means it rotates, thus delete it
+                            // if the file is uploaded and it is active file, means all files are uploaded
+                            if (startPosition == file.length()) {
```

Comment:
Use names that represent intent rather than comments. 

```java
boolean wasFileUploaded = startPosition == file.length();
boolean isNotActiveFile = !file.equals(activeFile);

if (wasFileUploaded && isNotActiveFile) {
}
```